File: app/src/main/java/com/softaai/designpatterns/java/creational/prototype/Square.java
package com.softaai.designpatterns.java.creational.prototype;

/**
 * Created by Amol Pawar on 01-04-2019.
 * softAai Apps
 */
public class Square extends Shape {

    public Square(){
        type = "Square";
    }
    @Override
    public void draw() {
        System.out.println("Inside Square::draw() method");
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/prototype/Square.kt
package com.softaai.designpatterns.kotlin.creational.prototype


/**
 * Created by Amol Pawar on 01-04-2019.
 * softAai Apps
 */
class Square : Shape() {
    init {
        shapeType = "Square"
    }
    override fun draw() {
        println("Inside Square::draw() method")
    }
}